src/meracache.jl

```diff
@@ -38,16 +38,17 @@ mutable struct MERACache{N, LT <: Layer, OT}
     # LT stands for Layer Type, OT for Operator Type.
     # TODO Should we use NTuple{N}s instead for densitymatrices and environments?
     densitymatrices::Vector{Union{Nothing, OT}}
-    operators::Dict{Any, Vector{OT}}
-    environments::Dict{Any, Vector{Union{Nothing, LT}}}
+    # probably a better choice to use IdDicts here
+    operators::IdDict{Any, Vector{OT}}
+    environments::IdDict{Any, Vector{Union{Nothing, LT}}}
     previous_fixedpoint_densitymatrix::Union{Nothing, OT}
     previous_operatorsum::Union{Nothing, OT}
 
     function MERACache{N, LT, OT}() where {N, LT, OT}
         @assert OT === operatortype(LT)
-        densitymatrices = Vector{Union{Nothing, OT}}(repeat([nothing], N))
-        operators = Dict{Any, Vector{OT}}()
-        environments = Dict{Any, Vector{Union{LT}}}()
+        densitymatrices = Vector{Union{Nothing, OT}}(fill(nothing, N))
+        operators = IdDict{Any, Vector{OT}}()
```

Comment:
The downside of `IdDict` is that if you create e.g. two copies of the same Hamiltonian, they won't end up picking up the same cache.

Reply:
Yes, I'll revert